core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,594 @@
+    # and we cannot rely on the backend updates to fully sync with the frontend
+    # report updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for. Possible categories include:
+    # suggestion_feature, suggestion_language, suggestion_other,
+    # issue_lesson_question, issue_general_language, issue_audio_language,
+    # issue_text_language, issue_topics, issue_profile, issue_other, crash.
+    category = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 0.1-alpha-abcdef1234) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The entry point location that the user is accessing the feedback report
+    # from on both web & Android devices. Possible entry points include:
+    # navigation_drawer, lesson_player, revision_card, or crash.
+    entry_point = datastore_services.StringProperty(required=True, indexed=True)
+    # Additional topic / story / exploration IDs that may be collected depending
+    # on the entry_point used to send the report; a lesson player entry point
+    # will have topic_id, story_id, and exploration_id, while revision cards
+    # will have topic_id and subtopic_id.
+    entry_point_topic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_story_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_exploration_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_subtopic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # The text language on Oppia set by the user in its ISO-639 language code;
+    # this is set by the user in Oppia's app preferences on all platforms.
+    text_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The audio language ISO-639 code on Oppia set by the user; this is set in
+    # Oppia's app preferences on all platforms.
+    audio_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    android_device_country_locale_code = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # The Android device model used to submit the report.
+    android_device_model = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # The Android SDK version on the user's device.
+    android_sdk_version = datastore_services.IntegerProperty(
+        required=False, indexed=True)
+    # The rest of the report info collected on Android; None if the platform is
```

Comment:
Don't frame this as "the rest"; instead, frame it as the Android-specific report info, which is its own concept in itself and should be constructively defined (rather than defined by exclusion). Ditto for web.

Reply:
Done